app/models/manageiq/providers/azure/event_catcher_mixin.rb

```diff
@@ -0,0 +1,38 @@
+module ManageIQ::Providers::Azure::EventCatcherMixin
+  def parse_event_type(event)
+    event_name = event["eventName"]["value"]
+    event_type = ""
+
+    unless event["authorization"].nil? || event["authorization"]["action"].nil?
+      event_type = parse_event_action(event)
+    end
+    "#{event_type}#{event_name}"
+  end
+
+  def parse_event_action(event)
+    # Example autorization action:
+    # Microsoft.Compute/virtualMachines/deallocate/action
+
+    action = event["authorization"]["action"].split("/")
+    "#{action[1]}_#{action[2]}_"
+  end
+
+  def parse_vm_ref(event)
+    # Example resourceId:
+    # /subscriptions/123456789-a1234-12b4-1234-5cd67890312/resourceGroups/
+    # rg_name/providers/Microsoft.Compute/virtualMachines/vm_name
+
+    resource_id = event["resourceId"].downcase.split("/")
```

Comment:
Would it be possible to name the parts instead of using numbered locations?  That is change this to something like

```ruby
_, _, vm_ref, _, resource_group_name, _, something, something_else, *_rest = event["resourceId"].downcase.split("/")
```

The names are terrible, but you get the idea :smile: